modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -56,7 +59,11 @@ object Rewards {
         events: Set[GlobalSnapshotEvent],
         maybeCalculatedState: Option[DataCalculatedState] = None
       ): F[SortedSet[RewardTransaction]] = {
-        val facilitators = lastArtifact.proofs.map(_.id)
+        val artifactFacilitators = lastArtifact.proofs.map(_.id)
+
+        val facilitators = seedlist.flatMap { s =>
+          artifactFacilitators.filter(id => s.contains(id.toPeerId)).toNes
+        }.getOrElse(artifactFacilitators)
```

Comment:
It is wrong imo because you are filtering out the node that actually signed `lastArtifact` but is not on the seedlist anymore. Imo we should allow rewarding historical commitment.